Dockerfile

```diff
@@ -1,39 +1,143 @@
-FROM node:12.14-alpine
+# ---------------------------------------------------------
+# Base build dependencies
+# ---------------------------------------------------------
+FROM node:12.14-alpine as builder-base
+
+WORKDIR /app
 
 # Install system dependencies
-# Add deploy user
 RUN apk --no-cache --quiet add \
       bash \
       build-base \
       curl \
-      dumb-init \
       git \
-      python && \
-      adduser -D -g '' deploy
+      python
 
-WORKDIR /app
+# ---------------------------------------------------------
+# Yarn base
+# ---------------------------------------------------------
+FROM builder-base as yarn-base
 
-# Set the correct owner on the /app
-RUN chown deploy:deploy $(pwd)
+# Copy files required for installation of application dependencies
+COPY package.json yarn.lock ./
+COPY patches ./patches
 
-# Switch to less-privileged user, do this early to ensure files are created
-# with proper ownership.
-USER deploy
+# ---------------------------------------------------------
+# Yarn dependencies
+# ---------------------------------------------------------
+FROM yarn-base as yarn-deps
 
-# Copy files required for installation of application dependencies
-COPY --chown=deploy:deploy package.json yarn.lock ./
-COPY --chown=deploy:deploy patches ./patches
+RUN yarn install --production --frozen-lockfile --quiet \
+  && mv node_modules /opt/node_modules.prod \
+  && yarn install --frozen-lockfile --quiet \
+  && yarn cache clean --force
 
-# Install application dependencies
-RUN yarn install --frozen-lockfile --quiet && \
-  yarn cache clean --force
+# ---------------------------------------------------------
+# Builder with source code
+# ---------------------------------------------------------
+FROM yarn-deps as builder-src
 
 # Copy application code
-COPY --chown=deploy:deploy . ./
+COPY __mocks__ ./__mocks__
+COPY cypress ./cypress
+COPY data ./data
+COPY patches ./patches
+COPY src ./src
+COPY webpack ./webpack
+COPY .env.oss \
```

Comment:
Do we access the `.env.oss` in tests? I wonder if it makes sense to consolidate those requirements in the `.env.test` files.

Reply:
I'm not sure I will test this. I remember adding it explicitly however that may have been before I added `.env.test`.